src/k8s/api/v1/cluster.go

```diff
@@ -14,6 +14,23 @@ type InitClusterRequest struct{}
 // InitClusterResponse is the response for "POST 1.0/k8sd/cluster".
 type InitClusterResponse struct{}
 
+// JoinClusterRequest is used to request the configuration to join the k8s cluster.
+type JoinClusterRequest struct {
+	Token string `json:"token"`
+}
+
+// JoinClusterResponse is the response for "POST 1.0/k8sd/cluster/join".
+type JoinClusterResponse struct {
+	ExtraServiceArgs ExtraServiceArgs `json:"service_argument_overwrites"`
```

Comment:
`extraServiceArgs` or `extra_service_args`, let's pick one and follow it everywhere.
```suggestion
	ExtraServiceArgs ExtraServiceArgs `json:"extraServiceArgs"`
```

Reply:
Personally, I am a fan of "named" arguments, e.g. `ExtraKubeletArgs`, `Extra$$$$Args`, etc, as it sets explicit expectations about field names and supported values, even though it leads to duplication.

I am happy to keep this if you consider too much work to change it.

Reply:
I agree. However, in this phase we probably need to play around a bit in terms of what parameters we want to overwrite or add. Having explicit parameters would mean significant overhead for a simple test (e.g. adding it to the response struct, parsing it etc.). I added a TODO to revise this once the k8s-snap matured a bit.